### PR TITLE
Handle queries where the EC2 subPath is "/", and misc fixes

### PR DESCRIPTION
**Pull request by ScottGarman**
We were reporting 404 errors when hitting the address

https://metadata.equinixmetal.net/2009-04-04/meta-data/

The trailing slash was routing us to a method that assumed subPath had
an endpoint in it. So account for an empty subPath by returning
metadata.ItemNames() as we would if the trailing slash was not included
in the request.

Update app and CI to use golang 1.20, misc cleanups.

### CODE REVIEW

File: pkg/api/v1/ec2_router.go [hunk 1]
```diff
@@ -50,6 +50,19 @@ func GetEc2MetadataItemPath(itemPath string) string {
 	return path.Join(V20090404URI, fmt.Sprintf("%s/%s", Ec2MetadataURI, trimmed))
 }
 
+// GetEc2MetadataItemPathWithoutTrim is used to test routing edge cases where
+// the trailing '/' is kept
+func GetEc2MetadataItemPathWithoutTrim(itemPath string) string {
```
Comment (mikemrm) [inline pkg/api/v1/ec2_router.go:55]: Does this need to be part of the apiv1 package? If this is only used for testing, this can be an unexported function within the file doing the testing.

Comment (ScottGarman) [reply pkg/api/v1/ec2_router.go]: Good point - I've now moved that into the test file, and eliminated some duplicated code as well.

File: pkg/api/v1/router_instance_ec2_metadata_test.go [hunk 2]
```diff
@@ -14,6 +14,20 @@ import (
 	v1api "go.hollow.sh/metadataservice/pkg/api/v1"
 )
 
+// GetEc2MetadataItemPathWithoutTrim is used to test routing edge cases where
+// the trailing '/' is kept
+func GetEc2MetadataItemPathWithoutTrim(itemPath string) string {
```
Comment (mikemrm) [inline pkg/api/v1/router_instance_ec2_metadata_test.go:19]: This shouldn't be exported.

**Review by mikemrm — APPROVED** [review]

File: pkg/api/v1/router_instance_ec2_metadata_test.go (hunk 2 above)
Comment (ScottGarman) [reply pkg/api/v1/router_instance_ec2_metadata_test.go]: Good catch - fixed!

**Review by mikemrm — APPROVED** [review]
